tests/RocksDBIteratorTest.cpp

```diff
@@ -0,0 +1,119 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+
+#include "iterlib/RocksDBIterator.h"
+
+#include <boost/filesystem.hpp>
+#include <folly/Singleton.h>
+#include <gtest/gtest.h>
+#include <memory>
+
+#include "iterlib/LimitIterator.h"
+#include "rocksdb/comparator.h"
+#include "rocksdb/db.h"
+#include "rocksdb/options.h"
+#include "rocksdb/table.h"
+
+using namespace rocksdb;
+std::string db_path;
+
+class RocksDBIteratorTest : public ::testing::Test {
+ public:
+  RocksDBIteratorTest() {}
+
+  rocksdb::DB* getDB() { return db_; }
+
+  rocksdb::Status Put(const Slice& k, const Slice& v,
+                      WriteOptions wo = WriteOptions()) {
+    return getDB()->Put(wo, k, v);
+  }
+
+  std::string IterStatus(Iterator* iter);
+
+  void SetUp() override {
+    db_path = boost::filesystem::unique_path().string();
+    Options options;
+    BlockBasedTableOptions tableOptions;
+    tableOptions.use_delta_encoding = false;
+    options.table_factory.reset(NewBlockBasedTableFactory(tableOptions));
+    options.comparator = ReverseBytewiseComparator();
+
+    options.create_if_missing = true;
+    options.error_if_exists = true;
+    Status status = DB::Open(options, db_path, &db_);
+    ASSERT_TRUE(status.ok());
+  }
+
+  void TearDown() override {
+    delete db_;
+    boost::filesystem::remove_all(db_path);
+  }
+
+ protected:
+  rocksdb::DB* db_;
+};
+
+std::string RocksDBIteratorTest::IterStatus(Iterator* iter) {
+  std::string result;
+  if (iter->Valid()) {
+    result = iter->key().ToString() + "->" + iter->value().ToString();
+  } else {
+    result = "(invalid)";
+  }
+  return result;
+}
+
+#define ASSERT_OK(s) ASSERT_TRUE((s).ok())
+#define P(s) folly::StringPiece(s)
+
+TEST_F(RocksDBIteratorTest, Simple) {
+  ASSERT_OK(Put("a", "1"));
+  ASSERT_OK(Put("b", "2"));
+  ASSERT_OK(Put("c", "3"));
+  ReadOptions ro;
+  ro.pin_data = true;
+  auto riter = getDB()->NewIterator(ro);
+  riter->SeekToFirst();
+  EXPECT_TRUE(riter->Valid());
+  auto iter = folly::make_unique<iterlib::RocksDBIterator>(riter);
+  iter->prepare();
+  EXPECT_TRUE(iter->next());
+  EXPECT_EQ(iterlib::Item(P("c")), iter->key());
```

Comment:
This found a bug in one of the tests (needed to set pin_data=true in both tests).
Thanks for suggesting.

Also, this code doesn't use ExpectIterator, because the code is operating at a lower level. Hope we can revisit in the future if necessary.